sdk/core/azure-core/src/main/java/com/azure/core/util/metrics/LongHistogram.java

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.util.metrics;
+
+import com.azure.core.util.AttributesBuilder;
+import com.azure.core.util.Context;
+
+/** A histogram instrument that records {@code long} values. */
+public interface LongHistogram {
+    /**
+     * Records a value with a set of attributes.
+     *
+     * @param value The amount of the measurement.
+     * @param attributes Collection of attributes representing metric dimensions.
+     * @param context The explicit context to associate with this measurement.
+     */
+    void record(long value, AttributesBuilder attributes, Context context);
```

Comment:
Consider:
- same naming here and in the LongCouter. I.e. both can be named `record` and behavior depends on what type is that. I.e. both LongCounter and LongHistogram are aggregators that aggregate in different way, so there might be abstract base for them (in the future perhaps).
- Is it safe to pass AttributeBuilder (I assume it's mutable and not necesarily thread safe)? should this accepet Attributes instead?

Reply:
> same naming here and in the LongCouter. I.e. both can be named record and behavior depends on what type is that. I.e. both LongCounter and LongHistogram are aggregators that aggregate in different way, so there might be abstract base for them (in the future perhaps).

OpenTelemetry and Micrometer don't have base counter types and they have different method names (perhaps to emphasize different effects these calls make - delta vs absolute value). I'm also not fully sure if having base class would make sense - we're not going to have collections of them or pass around without knowing what they are.

Reply:
The signatures look the same except name, but if that's the standard it's ok. 
How about the AttributeBuilder?

Reply:
how do the apis that take `TelemetryAttributes` are going to access values? It seems that `TelemetryAttributes` is kind of very simple immutable map. Should it be
```
public interface TelemetryAttributes {
 Object getAttribute(String key);
}
```
?

Reply:
Makes sense. Is it possible that user has multiple telemetry providers loaded? What happens if they try to pass `Tags` to OTel ?

Reply:
it's our SDK who'll pass tags/attributes around. I.e. they
- get specific `MeterProvider` with SPI, I believe the order in which multiple impls are returned is stable (since they are [cached](https://docs.oracle.com/javase/9/docs/api/java/util/ServiceLoader.html))
- use this provider to create meters and attributes and report measurements

I.e. all those interfaces are consistent with each other unless the package that implements them has a bug or we did something crazy in the SDK.

If something terribly unexpected happens, OTel impl will gracefully ignore attributes that it doesn't own and warn.

Reply:
Sounds good. I think empty interface should be ok then.